File: day16/primeFinder.cpp
#include <iostream>
#include <cmath>
#include <limits>

using namespace std;

bool isPrime(int num) {
    if (num <= 1) return false;
    for (int i = 2; i <= sqrt(num); ++i) {
        if (num % i == 0) return false;
    }
    return true;
}

void printPrimesUpTo(int n) {
    for (int i = 2; i <= n; ++i) {
        if (isPrime(i)) {
            cout << i << " ";
        }
    }
    cout << endl;
}

int getUserInput() {
    int num;
    while (true) {
        cout << "Enter a positive whole number: ";
        cin >> num;

        if (cin.fail() || num < 2) {
            cout << "Please enter a valid number (2 or greater)." << endl;
            cin.clear();
            cin.ignore(numeric_limits<streamsize>::max(), '\n');
        } else {
            break;
        }
    }
    return num;
}

int main() {
    int userNum = getUserInput();
    cout << "Prime numbers up to " << userNum << ":" << endl;
    printPrimesUpTo(userNum);
    return 0;
}

File: day16/prime_finder.rs
use std::io;

fn is_prime(n: u32) -> bool {
    if n <= 1 {
        return false;
    }
    for i in 2..=((n as f64).sqrt() as u32) {
        if n % i == 0 {
            return false;
        }
    }
    true
}

fn get_user_input() -> u32 {
    loop {
        println!("\nEnter a positive whole number (2 or greater):");
        let mut input = String::new();

        if io::stdin().read_line(&mut input).is_ok() {
            if let Ok(num) = input.trim().parse::<u32>() {
                if num >= 2 {
                    return num;
                }
            }
        }
        println!("Please enter a valid number (2 or greater).");
    }
}

fn print_primes_up_to(n: u32) {
    println!("\nPrime numbers up to {}:", n);
    for i in 2..=n {
        if is_prime(i) {
            print!("{} ", i);
        }
    }
    println!();
}

fn main() {
    let user_num = get_user_input();
    print_primes_up_to(user_num);
}
